src/main/java/com/kodilla/ecommercee/domain/Cart.java

```diff
@@ -0,0 +1,73 @@
+package com.kodilla.ecommercee.domain;
+
+import javax.persistence.*;
+import javax.validation.constraints.NotNull;
+
+@Entity
+@Table(name="CART")
+public class Cart {
+
+    private int id;
+    private String name;
+    private String description;
+    private double price;
+    private String groupId;
+
+    @Id
+    @GeneratedValue
+    @NotNull
+    @Column(name="ID", unique = true)
+    public int getId(){
+        return id;
+    }
+    @Column(name="NAME")
+    public String getName()
+    {return name;}
+
+    @Column(name="DESCRIPTION")
+    public String getDescription(){
+        return description;
+    }
+
+    @Column(name = "PRICE")
+    public double getPrice(){
+        return price;
+    }
+
+    @Column(name = "GROUP_ID")
+    public String getGroupId(){
+        return groupId;
```

Comment:
Takie powiązanie oznacza, że nie będzie relacji miedzy Cart, a Group, czy to na pewno dobry kierunek? Nie neguję, bardziej to temat do dyskusji. Obecnie, mając obiekt typu Cart, nie będzie można wykonać `getGroup`, które zwróci obiekt typu `Group`. Co zespół o tym myśli?

Reply:
Tu  dochodzimy do kwestii relacji pomiędzy encjami - schemat, który zaproponowałem na wiki ( https://github.com/pplutap/jdp-2005-03/wiki ) wcale nie musi (i zapewne nie jest) najlepszy, ba może nawet nie jest poprawny. Co do tej konkretnej sytuacji to rozumiem, że powiązanie pomiędzy Cart i Group będzie pozwalał np wyświetlić jakie grupy produktów znajdują się w koszyku. Dobrze myślę?

Reply:
dlatego tez w daily napisalem ze czekam z kolejnym zadaniem ze wzgledu na brak ustalen co do powiazan encji

Reply:
@MarcinSchmechel @tomysnatalia co myślicie?

Reply:
Ja chyba nie do końca rozumiem zależność między CART, PRODUCT i GROUP.  Chodzi o to, że mamy różne kategorie produktów, które można kupić, prawda? Nie wyobrażam sobie jednak, że w koszyku widzę grupy produktów. W koszyku powinienem widzieć produkty. To w jakich są one grupach jest drugorzędne. Z tego co widzę to w wg projektu Cart nie ma zwracać getGroup, czy źle patrzę? Endpointy kontrolera dotyczą Products. Czyli wg mnie nie ma relacji między Cart i Group.

Kuba, na schemacie wstawiłeś Products pomiędzy Cart i Product. Wytłumaczysz mi w jakim celu? Reszta schematu jest dla mnie zrozumiała i się z nią zgadzam, inna sprawa to jakie są relacje między tymi encjami. W większości oznaczyłeś je jako 1:1. Wg mnie:
-User może mieć wiele zamówień USER:ORDER 1:N
- ORDER i CART to 1:1
- koszyk może mieć wiele produktów CART:PRODUCT 1:N
- w grupie produktów mamy wiele produktów czyli GROUP:PRODUCT to 1:N 
(ewentualnie jeśli jeden produkt może być w kilku kategoriach to mamy N:N, ale proponuję wariant pierwszy)

Reply:
jeśli o mnie chodzi zgadzam w kwestii braku relacji pomiędzy Cart a Group. Uważam, że zbędna jest informacja o grupie produktu, który znajduje nie w koszyku.

Reply:
Również nie rozumiem do czego jest "Products". Wydaję mi się, że w Cart powinny być obiekty Product z zadaną ilością.
Co do Group to też nie widzę powiązania z Cart - przynajmniej dla użytkownika serwisu (od strony właściciela serwisu może być wymagany wgląd jakie grupy produktów sprzedawane są razem w jednym zamówieniu).

Reply:
Sformatowałem odpowiednio kod niestety jeśli chodzi o powiązania chciałbym je zmienić po środowej rozmowie.

Reply:
To ja mam taką propozycję - spotkanie jest w środę, żeby nie wstrzymywać prac - zróbmy wstępną implementację encji i dajmy je do dodania do gałęzi master - z takimi powiązaniami jakie każdy uważa dla swojej encji za prawidłowe - w środę przedyskutujemy całość powiązań i ewentualnie zmienimy tylko powiązania wskazane w encjach - co Wy na to? To wstrzyma testowanie encji ale pozwoli na ich wstępną implementację, co powinno zaoszczędzić trochę czasu.

Co do powyższych uwag to:

- macie rację USER do ORDER powinno być 1:N
- wstawiłem encję PRODUCTS bo wydawało mi się potrzebne z uwagi na specyfikację projektu w oparciu o json`y z katalogu specs - tu też się zgadzam, że nie ma sensu mnożyć bytów lepiej dać CART do PRODUCT 1:N
- również jak napisaliście GROUP do PRODUCT 1:N
- tak jak napisał Marcin - być może zamawiający serwis chce np. możliwości posortowania produktów w CART po grupach - pytanie jeśli PRODUCT jest powiązane z GROUP to ewentualne sortowanie może się odbywać poprzez powiązanie encji PRODUCT - nie ma potrzeby wiązania CART z GROUP.